qiskit/extensions/standard/u1.py

```diff
@@ -132,23 +131,54 @@ def _define(self):
           u1(lambda/2) b;
         }
         """
-        from qiskit.extensions.standard.x import CXGate
         definition = []
-        q = QuantumRegister(2, 'q')
-        rule = [
-            (U1Gate(self.params[0] / 2), [q[0]], []),
-            (CXGate(), [q[0], q[1]], []),
-            (U1Gate(-self.params[0] / 2), [q[1]], []),
-            (CXGate(), [q[0], q[1]], []),
-            (U1Gate(self.params[0] / 2), [q[1]], [])
-        ]
+        q = QuantumRegister(self.num_qubits, 'q')
+        if self.num_ctrl_qubits == 1:
+            from qiskit.extensions.standard.x import CXGate
+            rule = [
+                (U1Gate(self.params[0] / 2), [q[0]], []),
+                (CXGate(), [q[0], q[1]], []),
+                (U1Gate(-self.params[0] / 2), [q[1]], []),
+                (CXGate(), [q[0], q[1]], []),
+                (U1Gate(self.params[0] / 2), [q[1]], [])
+            ]
+        else:
+            rule = _mcu1_rule(q, self.params[0], self.num_ctrl_qubits)
+
         for inst in rule:
             definition.append(inst)
         self.definition = definition
 
+    def control(self, num_ctrl_qubits=1, label=None, ctrl_state=None):
+        """Controlled version of this gate.
+
+        Args:
+            num_ctrl_qubits (int): number of control qubits.
+            label (str or None): An optional label for the gate [Default: None]
+            ctrl_state (int or str or None): control state expressed as integer,
+                string (e.g. '110'), or None. If None, use all 1s.
+
+        Returns:
+            ControlledGate: controlled version of this gate.
+        """
+        if ctrl_state is None:
+            return CU1Gate(*self.params, num_ctrl_qubits=self.num_ctrl_qubits + num_ctrl_qubits)
+        return super().control(num_ctrl_qubits=num_ctrl_qubits, label=label,
+                               ctrl_state=ctrl_state)
+
     def inverse(self):
         """Invert this gate."""
-        return CU1Gate(-self.params[0])
+        return CU1Gate(-self.params[0], num_ctrl_qubits=self.num_ctrl_qubits)
+
+    def to_matrix(self):
+        """Return a numpy.array for the multi-controlled U1 gate."""
+        lam = self.params[0]
+        if self.num_ctrl_qubits == 0:
+            return U1Gate(lam).to_matrix()
+
+        from qiskit.extensions.unitary import _compute_control_matrix
+        base_mat = U1Gate(lam).to_matrix()
+        return _compute_control_matrix(base_mat, self.num_ctrl_qubits, ctrl_state=self.ctrl_state)
```

Comment:
Not sure whether the control state should be passed in here, other controlled gates do not support open controls in the matrix representation.

Reply:
Since the matrix should agree with the circuit `ctrl_state` should be added. Also, it may be good to generalize the open control test to go through all standard controlled gates to make sure this is done.